### PR TITLE
Reconfigured Interest Requests to use a name or title and find the id on server side

### PR DESCRIPTION
**Pull request by TheMkrage**
Now, a name or title is sent with the request to add or remove an interest. An id or ConvoInterst instance is sent back to the client (this is the id or object for the user or interest being added or removed) This method could be used with our next feature (send a name, get id from model, send id back to client)

### CODE REVIEW

File: src/codeu/chat/client/core/UserContext.java [hunk 1]
```diff
@@ -26,112 +26,96 @@
...
+
+	public UserContext(User user, BasicView view, BasicController controller) {
+		this.user = user;
+		this.view = view;
+		this.controller = controller;
+	}
+
+	public User addUserInterest(String name) {
+		Uuid idToAdd = controller.newUserInterest(user.id, name);
+		if (idToAdd == null || idToAdd.id() == user.id.id()) {
```
Comment (KevinWorkman) [inline src/codeu/chat/client/core/UserContext.java]: I would maybe add some comments to these type of functions. When will `idToAdd` be `null`? When will the IDs be equal? Similar comments in the other functions in this class.

File: src/codeu/chat/common/User.java [hunk 2]
```diff
@@ -33,226 +33,246 @@
...
 // ignore anything that may cause problems when converting to JSON
 // prevents crashes for items that cant be serialized.
-@JsonIgnoreProperties(ignoreUnknown=true)
+@JsonIgnoreProperties(ignoreUnknown = true)
 public final class User {
-  
-  public HashMap<Uuid, Integer> conversationInterests = new HashMap<>();
-  public ArrayList<Uuid> userInterests = new ArrayList<Uuid>();
+
+	public HashMap<Uuid, Integer> conversationInterests = new HashMap<>();
```
Comment (KevinWorkman) [inline src/codeu/chat/common/User.java]: Do these need to be public?

File: src/codeu/chat/common/User.java [hunk 3]
```diff
@@ -33,226 +33,246 @@
...
+	// arraylist keeps
+	// track of the conversations the userinterest has contributed messages to
+	// (CONVOS_CONTRIBUTED_TO_ARRAY)
+	//
+	public HashMap<Uuid, ArrayList<ArrayList<String>>> userStatusUpdate(
+			Iterable<ConversationContext> conversations, Time updateTime) {
+		// ASSUMPTION: all of the names saved in userInterests are indeed
+		// valid users in the system
+
+		HashMap<Uuid, ArrayList<ArrayList<String>>> userUpdates = new HashMap<Uuid, ArrayList<ArrayList<String>>>();
```
Comment (KevinWorkman) [inline src/codeu/chat/common/User.java:138]: Didn't Julia's PR change this so it uses a class instead?

**Review by KevinWorkman — APPROVED** [review]
In general, the code looks good. With a big change like this, this is exactly where unit tests come in handy: seeing that none of the tests break is good insurance that none of the code broke. But like we discussed before, it's not feasible to have unit tests for everything in this project.

Hopefully you at least did a ton of manual testing to make sure that everything still works how it used to: adding convos and messages, having multiple clients talking to teach other, adding interests, storing data between runs, etc. Might be good to include the results of those manual tests in the description?

Assuming that you've tested to make sure everything still works, I'll say this LGTM and is ready to be merged in!

File: src/codeu/chat/common/User.java (hunk 3 above)
Comment (TheMkrage) [reply src/codeu/chat/common/User.java:138]: Haven't pulled her changes in yet. Might jsut leave that for merge conflict when merging later

File: src/codeu/chat/common/User.java (hunk 2 above)
Comment (TheMkrage) [reply src/codeu/chat/common/User.java]: I don't think so!
updated

File: src/codeu/chat/client/core/UserContext.java (hunk 1 above)
Comment (TheMkrage) [reply src/codeu/chat/client/core/UserContext.java]: Comments added :)

**Comment by TheMkrage** [comment]
I did a lot of manual testing: adding removing interests and testing some corner cases like user adding themselves or convo that doesn't exist.
